app/javascript/packs/actions/options.js

```diff
@@ -0,0 +1,62 @@
+import {
+  FETCH_IN_PROGRESS,
+  OPTION_FETCH_IN_PROGRESS,
+  OPTION_SAVE_IN_PROGRESS,
+  PRE_FETCH_SAVE,
+  CREATE_UNSAVED_OPTION,
+  FETCH_FAILURE,
+  FETCH_SUCCESS,
+  OPTION_FETCH_FAILURE,
+  OPTION_FETCH_SUCCESS,
+  REMOVE_OPTION
+} from '../constants';
+
+export function optionFetchInProgress(option) {
+  return {
+    type: OPTION_FETCH_IN_PROGRESS,
+    status: FETCH_IN_PROGRESS,
+    fetching: true,
+    option
+  }
+}
+
+export function optionPreFetchSave(option) {
+  return {
+    type: OPTION_SAVE_IN_PROGRESS,
+    fetching: false,
+    status: PRE_FETCH_SAVE,
```

Comment:
@kevintli This is the `FETCH_NEEDED` constant we created today. Want to use this constant instead in #180?

Reply:
Sounds good! I can change it once this branch gets merged in